### DISCUSSION TITLE
How to convert Revit model with element color into iModel with Snapshot tool.

### DISCUSSION THREAD

**Discussion by Simplefair** [iTwin/itwinjs-core#684] [Q&A] [answered]
Sir, 

I try to convert Revit MEP model into imodel with Snapshot. I already set the element color such as Pipe with Blue, but I can't see the element color after I get .bim file. How I config Snapshot environment to get the Revit element color setting and display them in .bim file.

Thanks a advance.

**Comment by bbastings** [comment, marked as answer]
> I already set the element color such as Pipe with Blue,

How was this done? For revit data, the element color would typically come from the SubCategoryAppearance.

https://www.itwinjs.org/learning/common/geometrystream/
https://github.com/imodeljs/imodeljs/blob/master/core/common/src/SubCategoryAppearance.ts

**Reply by Simplefair** [reply]
Thank you for quick response. I will try it.

**Reply by paprophet** [reply]
> > I already set the element color such as Pipe with Blue,
> 
> How was this done? For revit data, the element color would typically come from the SubCategoryAppearance.
> 
> https://www.itwinjs.org/learning/common/geometrystream/
> https://github.com/imodeljs/imodeljs/blob/master/core/common/src/SubCategoryAppearance.ts

  category.setDefaultAppearance(new SubCategoryAppearance({ color: 0xff0000 }));

but  how get the category object by imodeljs frontend?

**Comment by pmconne** [comment]
>  I can't see the element color after I get .bim file.

Explain what you mean. Do you see the element, but it's a different color than you expect? Or do you not see the element at all? If you provide details, @abeesh ought to be able to find someone who can help.

**Reply by Simplefair** [reply]
I can see the element color in Revit but can't see them after I convert them into .BIM. I think maybe the problem is we don't know what kind of element color in Revit can be convert by imodel Snapshot and how to define the element color in Revit.

**Reply by pmconne** [reply]
@abeesh please assist @Simplefair with configuring iTwinSnapshot to properly convert element color from Revit files.

**Comment by abeesh** [comment]
Was the color assigned to the Revit element in the 3D view?  Can you please share the Revit file with us?

**Reply by Simplefair** [reply]
Yes Sir. I can share the Revit file to you, but the Zip file is almost 40 MB. How I will send it to you?

**Comment by Simplefair** [comment]
[
![Revit](https://user-images.githubusercontent.com/42367105/106724486-3c926080-6643-11eb-857c-9d8f4755be6d.jpg)
![Snapshot](https://user-images.githubusercontent.com/42367105/106724506-41efab00-6643-11eb-8c87-fc43d55b494b.jpg)




](url)

**Reply by Simplefair** [reply]
The upload file size is limited to 10MB. How I can send the revit file to you?

Thank you very much.

**Comment by Simplefair** [comment]
Two screen shot. One is from Revit and another is from imodel.

**Comment by abeesh** [comment]
The Revit connector uses physical material colors based on Realistic mode. Can you try changing the material of the elements or change the material’s texture/color?

![image](https://user-images.githubusercontent.com/3654177/106758610-5312ca80-65f7-11eb-8901-1aea5483ec37.png)

Please upload the Revit file to 
https://bentley.sharefile.com/r-r3d1598806d214cbca4cb5e3fef6e45ff

**Reply by Simplefair** [reply]
Abeesh,

Thank for your quick response. I had uploaded the Revit based on your web link. 

Thanks for you help.

Jianhui

**Reply by Simplefair** [reply]
Sir,

I do convert Revit model based on Realistic mode. I can see the element color in Revit but can't see them after I upload to imodel sever.

So any update for my uploaded revit file?

Thank you very much!

Jianhui

**Reply by timlawrence-bentley** [reply]
Hi Jianhui,

Thanks for the file. We have taken a look at the file and we have found a few issues with handling of colour overrides that we have not encountered before. These issues can be split into parts that we can fix on our iModel Connector and there are also issues that come from a 3rd party library. We hope to complete some of the fixes on our end in a few weeks, but full support for will take a bit longer than that on both our end as well as getting a fix from the 3rd party library.

Thanks,

Tim

**Reply by Simplefair** [reply]
Hi Tim,

Thanks for your response. I am really expecting these fixes. When you done, could you inform me please because there is no upgrade information about each version.

By the way, is there any technical document about how to set up Revit environment and optimize the revit file in order to  make it easy to convert to BIM file.  

Thanks you very much and happy Chinese New Year.

Jianhui

**Reply by pmconne** [reply]
@timlawrence-bentley @abeesh any updates?
